spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1038,6 +1038,20 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+  test("hex") {
+    Seq(true, false).foreach { dictionaryEnabled =>
+      withTempDir { dir =>
+        val path = new Path(dir.toURI.toString, "hex.parquet")
+        makeParquetFileAllTypes(path, dictionaryEnabled = dictionaryEnabled, 10000)
+
+        withParquetTable(path.toString, "tbl") {
+          // _9 and _10 (uint8 and uint16) not supported
+          checkSparkAnswerAndOperator(
+            "SELECT hex(_1), hex(_2), hex(_3), hex(_4), hex(_5), hex(_6), hex(_7), hex(_8), hex(_11), hex(_12), hex(_13), hex(_14), hex(_15), hex(_16), hex(_17), hex(_18), hex(_19), hex(_20) FROM tbl")
```

Comment:
Thanks, I agree it's best to test since they're there. I just genuinely thought it wasn't supported given the test coverage, the errors on the other functions, etc. I'll have a look at adding some tests...

Reply:
I'm realizing that I'm not going to have time next week and didn't expect this PR to take this long, so I've removed the scalar handling for now and hopefully can follow and example in the future. https://github.com/apache/datafusion-comet/pull/449/commits/88bdcde1babcd75cb52e18164e1c945bbbbbf857

Reply:
I initially tried and got test failures because the native function isn't being recognized w/ scalars. The same issue I mentioned w/ `trim` above.

Reply:
Sounds like QueryPlanSerde is failing the case match. 
Wondering if you already have tried `"spark.sql.optimizer.excludedRules" -> "org.apache.spark.sql.catalyst.optimizer.ConstantFolding"` ?

Also right now, since the scalar code was removed from rust, it will fail if the native code happen to find scalar values...

Reply:
Thanks, yes, I did also try disabling ConstantFolding to no avail.

Reply:
`hex(literal)` should be evaluated to literal early in Spark optimizer. I wonder what test failure you saw?